cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java

```diff
@@ -100,11 +101,14 @@ public void execute(Project project) {
             if (!project.currentPackage().getResolution().diagnosticResult().hasErrors()) {
                 if (!project.kind().equals(ProjectKind.BALA_PROJECT)) {
                     // BalaProject is a read-only project.
-                    // Hence we run the code generators/ modifiers only for BuildProject and SingleFileProject
-                    DiagnosticResult codeGenAndModifyDiagnosticResult = project.currentPackage()
-                            .runCodeGenAndModifyPlugins();
-                    if (codeGenAndModifyDiagnosticResult != null) {
-                        diagnostics.addAll(codeGenAndModifyDiagnosticResult.diagnostics());
+                    // Hence, we run the code generators/ modifiers only for BuildProject and SingleFileProject
+                    if (isProjectUpdated(project)) {
```

Comment:
using this multiple times can result in addition to build time since we check multiple files. We can do the check once in the Command Class and pass it on to the tasks. You can modify the constructor for this.

Reply:
Added to `CompileTask` constructor and used it inside.